src/commands/init.ts

```diff
@@ -82,160 +119,215 @@ export abstract class InitCommand extends Command {
           }
-          architect_service.build.context = service.build.context;
-          if (service.build.dockerfile) {
-            architect_service.build.dockerfile = service.build.dockerfile;
+          if (converted_props.base) {
+            (architect_service as any)[architect_property_name] = converted_props.base;
           }
+        } else {
+          this.log(chalk.yellow(`Could not convert ${service_name} property ${property_name}`));
         }
       }
+      architect_component.services[service_name] = architect_service;
+    }
 
-      let port_index = 0;
-      architect_service.interfaces = {};
-      for (const port of service.ports || []) {
-        if (typeof port === 'string' || typeof port === 'number') {
-          const single_number_port_regex = new RegExp('^\\d+$');
-          const single_port_regex = new RegExp('(\\d+[:]\\d+)\\/*([a-zA-Z]+)*$');
-          const port_range_regex = new RegExp('(\\d+[-]\\d+)\\/*([a-zA-Z]+)*$');
-
-          if (single_number_port_regex.test(port)) {
-            architect_service.interfaces[`interface${port_index}`] = port;
-            port_index++;
-          } else if (single_port_regex.test(port)) {
-            const matches = single_port_regex.exec(port);
-            const interface_spec: Partial<ServiceInterfaceSpec> = {};
-            if (matches && matches.length >= 3) {
-              interface_spec.protocol = matches[2];
-            }
-            if (matches && matches.length >= 2) {
-              interface_spec.port = parseInt(matches[1].split(':')[1]);
-            }
-            (architect_service.interfaces[`interface${port_index}`] as Partial<ServiceInterfaceSpec>) = interface_spec;
-            port_index++;
-          } else if (port_range_regex.test(port)) {
-            const matches = port_range_regex.exec(port);
-            if (matches && matches.length >= 2) {
-              const [start, end] = matches[1].split('-');
-              for (let i = parseInt(start); i < parseInt(end) + 1; i++) {
-                architect_service.interfaces[`interface${port_index}`] = i;
-                port_index++;
-              }
-            }
+    for (const service_config of Object.values(architect_component.services || {})) {
+      for (const depends_on of (service_config.depends_on || [])) {
+        service_config.environment = service_config.environment || {};
+        if (Object.keys(architect_component.services[depends_on].interfaces || {}).length) {
+          service_config.environment[`${depends_on.replace('-', '_').toUpperCase()}_URL`] = `\${{ services.${depends_on}.interfaces.main.url }}`;
+        }
+      }
+    }
+
+    let architect_yml = yaml.dump(yaml.load(JSON.stringify(classToPlain(architect_component))));
+    const local_regex = new RegExp(`${this.local_config_placeholder}:`, 'gm');
+    architect_yml = architect_yml.replace(local_regex, "${{ if architect.environment == 'local' }}:");
+    validateOrRejectSpec(yaml.load(architect_yml));
```

Comment:
If for some reason the generated architect.yml file wouldn't be valid, the user would get one or more validation errors that look like this:

```
ValidationErrors
[
  {
    "component": "architect-ryan/test",
    "path": "nonsense",
    "message": "Invalid key: nonsense - Did you mean tasks?",
    "value": "bad_value",
    "invalid_key": true
  }
]
```

I get that that can still be confusing though since it could be a bad conversion within our code. Should we create some kind of generalized error text for the CLI? `catch` in `base-command.ts` seems like a good place to put something like that

Reply:
I think this is a special case though. To be clear I am more worried about our code doing a poor conversion than something in the docker compose being nonsense. 

We should still present the user that information, but we should also give them recourse. At worst we should provide a link to our issue tracker to report a bug.